backend/src/AppServer.hs

```diff
@@ -375,13 +386,48 @@ exportHandler = do
 
     pure (players, gameReports, playerStatsYears, playerStatsTotals, playerStatsInits, leaguePlayers)
 
-  let entries =
-        [ toEntry "players.csv" 0 . encode $ players,
-          toEntry "gameReports.csv" 0 . encode $ gameReports,
-          toEntry "playerStatsYears.csv" 0 . encode $ playerStatsYears,
-          toEntry "playerStatsTotals.csv" 0 . encode $ playerStatsTotals,
-          toEntry "playerStatsInits.csv" 0 . encode $ playerStatsInits,
-          toEntry "leaguePlayers.csv" 0 . encode $ leaguePlayers
+  let exportPlayers = entityVal <$> players
+      playerNameMap = Map.fromList $ (\(Entity pid player) -> (pid, player.playerDisplayName)) <$> players
+      lookupPlayerName pid = Map.findWithDefault "(unknown player)" pid playerNameMap
+      exportGameReports =
+        gameReports <&> \report@GameReport {..} ->
+          ExportGameReport
+            { exportGameReportRecord = report,
+              exportGameReportWinnerName = lookupPlayerName gameReportWinnerId,
+              exportGameReportLoserName = lookupPlayerName gameReportLoserId
+            }
```

Comment:
This ended up great, the code changes are organized and easy to follow. I like the `export` layer and that it has flexibility to capture more data associations in the future if that ever came up. I don't give AI enough credit, impressed!

Reply:
*Some* of that was me post facto, but overall I agree with the sentiment. For every good choice, AI seems to make a really dumb one, but it can be a time saver to at least check what it has to say.